File: SubletUC.Core/Models/Listing.cs
namespace SubletUC.Server.Models
{
    public class Listing
    {
        public int ListingId { get; set; }
        public string? Address { get; set; }
        public decimal Rent { get; set; }
        public string? Availability { get; set; }
        public int Bedrooms { get; set; }
        public int Bathrooms { get; set; }
        public string? Description { get; set; }
        public bool UtilitiesIncludedInRent { get; set; }
        public decimal AverageUtilities { get; set; }
        public bool SharedRoom { get; set; }
        public int SharedRoommates { get; set; }
        public bool CatsAndDogsAllowed { get; set; }
        public bool WasherDryer { get; set; }
        public bool OffStreetParking { get; set; }
        public bool Driveway { get; set; }

        public decimal DistanceFromCampus { get; set; }

        public string? Notes { get; set; }

        public byte[]? Photo { get; set; }
        // Navigation property for related Roommates
        public ICollection<Roommate>? Roommates { get; set; }
        public bool Deleted { get; set; }
    }
}

File: SubletUC.client/src/app/data/listing.ts
import { Roommate } from "./enum/roommate";

export interface Listing {
    listingId: number
    address: string,
    rent: number,
    availability: string,
    bedrooms: number,
    bathrooms: number,
    description: string
    utilitiesIncludedInRent: boolean,
    averageutilities: number,
    roommates: Roommate[],
    sharedRoom: boolean,
    sharedRoommates: number,
    catsAndDogsAllowed: boolean,
    washerDryer: boolean,
    offStreetParking: boolean,
    driveway: boolean,
    distanceFromCampus: number, //walking distance
    notes: string, //additional notes worth mentioning
    photo?: File,
    deleted?: boolean,
    userId: number, //use as foreign key (one user => many listings)
}